File: 38-Java/src/Veicolo.java
import java.util.LinkedList;
import java.util.List;

public class Veicolo {
    protected int codice, postiLetto;
    protected String tipo, marca;
    protected float lunghezza, larghezza, costoGiornaliero;
    protected List<Prenotazione> prenotazioni;
    
    public Veicolo(int codice, int postiLetto, String tipo, String marca, float lunghezza, float larghezza,
            float costoGiornaliero) {
        this.codice = codice;
        this.postiLetto = postiLetto;
        this.tipo = tipo;
        this.marca = marca;
        this.lunghezza = lunghezza;
        this.larghezza = larghezza;
        this.costoGiornaliero = costoGiornaliero;
        prenotazioni = new LinkedList<Prenotazione>();
    }
    
    public void addPrenotazione(Prenotazione p){
        prenotazioni.add(p);
    }

    public float incasso(){
        int giorni = 0;
        for (Prenotazione p : prenotazioni)
            giorni = giorni + p.getGiorni();
        return giorni * costoGiornaliero;
    }

    public int getCodice() {
        return codice;
    }

    public String toString(){
        return tipo + "\t" + codice + "\t" + marca + "\t" + larghezza + "\t" + lunghezza + "\t" + postiLetto;
    }
}

File: 38-Python/Veicolo.py
class Veicolo:
    def __init__(self, codice, marca, larghezza, lunghezza, costo, posti) -> None:
        self._codice = codice
        self._marca = marca
        self._larghezza = larghezza
        self._lunghezza = lunghezza
        self._costo = costo
        self._posti = posti
        self._noleggi = []

    def addNoleggio(self, noleggi):
        self._noleggi.append(noleggi)

    def incasso(self):
        giorni = 0
        for n in self._noleggi:
            giorni = giorni + n.getGiorni()
        return giorni * self._costo

    def getCodice(self):
        return self._codice
    
    def __str__(self) -> str:
        return str(self._codice) + "\t" + self._marca + "\t" + str(self._larghezza) + "\t" + str(self._lunghezza) + "\t" + str(self._posti)